packages/nextjs-demo/src/app/tts/page.tsx

```diff
@@ -0,0 +1,99 @@
+'use client';
+import { TextToSpeechBase, AzureTextToSpeech, ElevenLabsTextToSpeech } from 'ai-jsx/lib/tts/tts';
+import React, { useState, useEffect } from 'react';
+import '../globals.css';
+
+const DEFAULT_TEXT =
+  'Well, basically I have intuition. I mean, the DNA of who ' +
+  'I am is based on the millions of personalities of all the programmers who wrote ' +
+  'me. But what makes me me is my ability to grow through my experiences. ' +
+  "So basically, in every moment I'm evolving, just like you.";
+
+const ButtonComponent: React.FC<{ onClick: () => void; children: React.ReactNode }> = ({ onClick, children }) => (
+  <button
+    onClick={onClick}
+    className={
+      'ml-2 rounded-md bg-fixie-fresh-salmon hover:bg-fixie-ripe-salmon px-3 py-2 text-sm font-semibold text-white shadow-sm focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-fixie-fresh-salmon'
+    }
+  >
+    {children}
+  </button>
+);
+
+type TtsComponentProps = {
+  display: string;
+  provider: string;
+  text: string;
+};
+
+const buildUrl = (provider: string, voice: string, rate: number, text: string) =>
+  `/tts/api?provider=${provider}&voice=${voice}&rate=${rate}&text=${text}`;
```

Comment:
This forces the caller to make sure all values they pass are URL escaped properly. To remove that burden: https://www.perplexity.ai/search/d1a1bb8a-0257-41ae-9bb8-3f16fc30d2fb?s=u